File: src/main/java/mandarin/packpack/commands/SendMessage.java
package mandarin.packpack.commands;

import common.CommonStatic;
import mandarin.packpack.supporter.server.CommandLoader;
import mandarin.packpack.supporter.server.data.IDHolder;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.channel.middleman.GuildChannel;
import net.dv8tion.jda.api.entities.channel.middleman.GuildMessageChannel;
import net.dv8tion.jda.api.entities.channel.middleman.MessageChannel;
import net.dv8tion.jda.api.sharding.ShardManager;
import javax.annotation.Nonnull;

public class SendMessage extends ConstraintCommand {
    public SendMessage(ROLE role, CommonStatic.Lang.Locale lang, IDHolder id) {
        super(role, lang, id, false);
    }

    @Override
    public void doSomething(@Nonnull CommandLoader loader) {
        MessageChannel ch = loader.getChannel();

        ShardManager client = ch.getJDA().getShardManager();

        if (client == null)
            return;

        String[] contents = loader.getContent().split(" ", 4);

        if(contents.length != 4) {
            ch.sendMessage("`p!sm [Guild ID] [Channel ID] [Contents]`").queue();
            return;
        }

        Guild g = client.getGuildById(contents[1]);

        if(g != null) {
            GuildChannel c = g.getGuildChannelById(contents[2]);

            if(c != null) {
                if(c instanceof GuildMessageChannel) {
                    String co;

                    if(contents[3].startsWith("`"))
                        co = contents[3].substring(1, contents[3].length() - 1).replace("\\e", "");
                    else
                        co = contents[3].replace("\\e", "");

                    ((GuildMessageChannel) c).sendMessage(co).queue();

                    ch.sendMessage(co).queue();

                    System.out.println(co);
                } else {
                    ch.sendMessage("Channel isn't message channel").queue();
                }
            } else {
                ch.sendMessage("No such channel found").queue();
            }
        } else {
            ch.sendMessage("No such guild found").queue();
        }
    }
}

File: src/main/java/mandarin/card/commands/SendMessage.kt
package mandarin.card.commands

import common.CommonStatic
import mandarin.card.supporter.ServerData
import mandarin.packpack.commands.Command
import mandarin.packpack.supporter.StaticStore
import mandarin.packpack.supporter.server.CommandLoader
import net.dv8tion.jda.api.entities.channel.middleman.GuildMessageChannel

class SendMessage : Command(CommonStatic.Lang.Locale.EN, false) {
    override fun doSomething(loader: CommandLoader) {
        val u = loader.user

        if (u.id != StaticStore.MANDARIN_SMELL && u.id != ServerData.get("gid")) {
            return
        }

        val contents = loader.content.split(Regex(" "), 4)

        if (contents.size != 4) {
            replyToMessageSafely(loader.channel, "`cd.sm [Guild ID] [Channel ID] [Content]`", loader.message) { a -> a }

            return
        }

        val g = loader.client.getGuildById(contents[1]) ?: return

        val channel = g.getGuildChannelById(contents[2]) ?: return

        if (channel !is GuildMessageChannel)
            return

        val content = if (contents[3].startsWith("`"))
            contents[3].substring(1, contents[3].length - 1).replace(Regex("\\e"), "")
        else
            contents[3].replace(Regex("\\e"), "")

        channel.sendMessage(content).queue()
        replyToMessageSafely(loader.channel, content, loader.message) { a -> a }
    }
}